### PR TITLE
[DROOLS-7638]"docker system prune -f" fails in GHA

### PR DESCRIPTION
**Pull request by tkobayas**
**JIRA**: 
https://issues.redhat.com/browse/DROOLS-7638

### CODE REVIEW

**Comment by tkobayas** [comment]
Hmm,
```
      current: |
        bash -c "if docker system df > /dev/null 2>&1; then docker system prune -f; else echo 'Prune operation already in progress. Skipping this command.'; fi"
```
resulted in
```
2024-10-17T05:15:51.3981823Z [command]/usr/bin/bash -c if docker system df > /dev/null 2>&1; then docker system prune -f; else echo 'Prune operation already in progress. Skipping this command.'; fi
2024-10-17T05:15:51.4689072Z Error response from daemon: a prune operation is already running
```

It means `docker system df` is not effective to check the lock.

We may just proceed when we hit "a prune operation is already running".

**Comment by tkobayas** [comment]
Okay. In Kogito Downstream / kogito-apps,
```
2024-10-17T06:22:00.8930991Z [command]/usr/bin/bash -c docker system prune -f 2>&1 | grep -q 'a prune operation is already running' && echo 'Prune operation already in progress. Skipping this command.' || true
2024-10-17T06:22:00.9090193Z Prune operation already in progress. Skipping this command.
```
GHA skips `docker system prune -f` in this case. I think it's a safe workaround, although we haven't identified why kogito-apps hits this issue.

File: .ci/buildchain-config.yaml [hunk 1]
```diff
@@ -25,7 +25,7 @@ default:
       mvn clean install -Dquickly ${{ env.BUILD_MVN_OPTS }} ${{ env.BUILD_MVN_OPTS_UPSTREAM }}
     after:
       current: |
-        docker system prune -f
+        bash -c "docker system prune -f 2>&1 | grep -q 'a prune operation is already running' && echo 'Prune operation already in progress. Skipping this command.' || true"
```
Comment (mareknovotny) [inline .ci/buildchain-config.yaml]: maybe better to use different command than `prune`?

see this for instance https://github.com/docker/for-mac/issues/2501
As recommended `docker rmi $(docker images -f "dangling=true"-q)` is logging the progress

File: .ci/buildchain-config.yaml [hunk 2]
```diff
@@ -25,7 +25,7 @@ default:
       mvn clean install -Dquickly ${{ env.BUILD_MVN_OPTS }} ${{ env.BUILD_MVN_OPTS_UPSTREAM }}
     after:
       current: |
-        docker system prune -f
+        bash -c "if [ -n \"$(docker images -f dangling=true -q)\" ]; then docker rmi $(docker images -f dangling=true -q); else echo 'No dangling images to remove.'; fi"
```
Comment (tkobayas) [inline .ci/buildchain-config.yaml:28]: Used `if` block because `docker rmi` fails if there is no dangling images.

**Comment by tkobayas** [comment]
@mareknovotny Thanks, replaced with `docker rmi $(docker images -f dangling=true -q)`.

```
2024-10-18T06:36:13.1706053Z [command]/usr/bin/bash -c if [ -n "$(docker images -f dangling=true -q)" ]; then docker rmi $(docker images -f dangling=true -q); else echo 'No dangling images to remove.'; fi
2024-10-18T06:36:13.1893638Z No dangling images to remove
```

Actually, there is no dangling images in this case. But I think it's good to go.

**Comment by tkobayas** [comment]
@rgdoliveira @mareknovotny Please review, thanks!

**Review by rgdoliveira — APPROVED** [review]

**Pull request merged**